### PR TITLE
Define NetApp E-Series ASUP module

### PR DESCRIPTION
**Pull request by lmprice**
##### SUMMARY

ASUP, or Auto-Support, is a mechanism that allows NetApp support
to receive information on the status of E-Series storage-systems
in order to proactively resolve issues for customers. This module
gives customers the ability to tune ASUP settings to their liking,
including disabling the feature entirely.

##### ISSUE TYPE

- New Module Pull Request
##### COMPONENT NAME

netapp_e_asup
##### ANSIBLE VERSION

```
ansible 2.6.0 (detached HEAD ce8c336d52) last updated 2018/04/18 11:23:03 (GMT -500)
  config file = None
  configured module search path = [u'/home/lorenp/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/lorenp/ansible/lib/ansible
  executable location = /home/lorenp/ansible/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by hulquest** [comment]
LGTM
shipit

File: test/integration/integration_config.yml [hunk 1]
```diff
@@ -1,3 +0,0 @@
----
```
Comment (mattclay) [inline test/integration/integration_config.yml]: Why was this file removed?

Comment (lmprice) [reply test/integration/integration_config.yml]: Ah, that's the reason why the tests failed. I was really intending to clear out any changes we made to the file. Oops. Our local CI has an independent copy, which is why I didn't see any issues before pushing.

**Comment by ndswartz** [comment]
shipit

**Review by ndswartz — APPROVED** [review]
shipit

File: lib/ansible/utils/module_docs_fragments/netapp.py [hunk 2]
```diff
@@ -121,24 +121,31 @@ class ModuleDocFragment(object):
   api_username:
     required: true
     description:
-    - The username to authenticate with the SANtricity WebServices Proxy or embedded REST API.
+    - The username to authenticate with the SANtricity Web Services Proxy or Embedded Web Services API.
```
Comment (gundalow) [inline lib/ansible/utils/module_docs_fragments/netapp.py]: I think this change needs removing from this PR (as it's been done in another PR which has since been merged)

Also you need to rebase to resolve the merge conflict https://docs.ansible.com/ansible/devel/dev_guide/developing_rebasing.html

File: test/sanity/validate-modules/ignore.txt [hunk 3]
```diff
@@ -1138,7 +1138,6 @@ lib/ansible/modules/storage/netapp/na_cdot_volume.py E322
 lib/ansible/modules/storage/netapp/na_cdot_volume.py E324
 lib/ansible/modules/storage/netapp/na_cdot_volume.py E325
 lib/ansible/modules/storage/netapp/netapp_e_amg.py E322
-lib/ansible/modules/storage/netapp/netapp_e_amg.py E325
 lib/ansible/modules/storage/netapp/netapp_e_amg_role.py E322
```
Comment (gundalow) [inline test/sanity/validate-modules/ignore.txt]: I think this change needs removing from this PR (as it's been done in another PR which has since been merged)

Also you need to rebase to resolve the merge conflict https://docs.ansible.com/ansible/devel/dev_guide/developing_rebasing.html

File: lib/ansible/modules/storage/netapp/netapp_e_asup.py [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+    - netapp.eseries
+options:
+    state:
+        description:
+            - Enable/disable the E-Series auto-support configuration.
+            - When this option is enabled, configuration, logs, and other support-related information will be relayed
+              to NetApp to help better support your system. No personally identifiable information, passwords, etc, will
+              be collected.
+        default: present
+        choices:
```
Comment (gundalow) [inline lib/ansible/modules/storage/netapp/netapp_e_asup.py]: enabled/disabled only

File: lib/ansible/modules/storage/netapp/netapp_e_asup.py [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+            - asup
+            - auto_support
+            - autosupport
+    active:
+        description:
+            - Enable active/proactive monitoring for ASUP. When a problem is detected by our monitoring systems, it's
+              possible that the bundle did not contain all of the required information at the time of the event.
+              Enabling this option allows NetApp support personnel to manually request transmission or re-transmission
+              of support data in order ot resolve the problem.
+            - Only applicable if I(state=present).
```
Comment (gundalow) [inline lib/ansible/modules/storage/netapp/netapp_e_asup.py]: enabled

File: lib/ansible/modules/storage/netapp/netapp_e_asup.py [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+
+
+class Asup(object):
+    DAYS_OPTIONS = ['sunday', 'monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday']
+
+    def __init__(self):
+        argument_spec = eseries_host_argument_spec()
+        argument_spec.update(dict(
+            state=dict(type='str', required=False, default='present', aliases=['asup', 'auto_support', 'autosupport'],
+                       choices=['present', 'absent', 'enabled', 'disabled']),
```
Comment (gundalow) [inline lib/ansible/modules/storage/netapp/netapp_e_asup.py]: enabled/disabled only

**Review by gundalow — CHANGES_REQUESTED** [review]

**Comment by lmprice** [comment]
@gundalow I believe I've resolved all of the requested changes with this patch.

**Comment by lmprice** [comment]
@gundalow The requested changes have been resolved.

**Review by gundalow — APPROVED** [review]
